### ISSUE TITLE
Support CamelCase in routing was broken in 21.3

### ISSUE THREAD

**Issue by matemax** [sanic-org/sanic#2084]
**Describe the bug**
backward compatibility was breaking in 21.3. Route regexp param name forbided to use capital leters [regexp here](https://github.com/sanic-org/sanic-router/blob/master/sanic_routing/patterns.py#L10). If project use  CamelCase for variable naming the update to  version  21.3 will broken app.

**Expected behavior**

Support CamelCase and snake_case.

**Environment (please complete the following information):**
 - Version 21.3

**Comment by ahopkins** [comment]
Can you give me an example? I am not sure I understand.

Do you mean this:

```python
@app.get("/<Foo>")
def foo(request):
    return text("foo")
```

**Comment by matemax** [comment]
@ahopkins Yes, it is my case. I created [PR](https://github.com/sanic-org/sanic-router/pull/10)

**Comment by ahopkins** [comment]
FYI - released in `sanic-routing==0.4.1.` Should be included with fresh installs of `sanic==21.3.1`.

```
pip install --upgrade sanic
```

**Comment by matemax** [comment]
thanx
